TUTORIAL.md

```diff
@@ -0,0 +1,100 @@
+# Getting started
+Before you start, make sure you understand the concept or database migration and rollback.
+
+Create new Go project and configure a database for it. Make sure that your database is supported [here](README.md#databases)
+For the purpose of this tutorial let's create PostgreSQL database called `example`.
+Our user here is `postgres`, and host is `localhost`.
+```
+psql -h localhost -U postgres -w -c "create database example;"
+```
+When using Migrate CLI we need to pass to database URL. Let's export it to a variable for convienience:
+```
+export POSTGRESQL_URL=postgres://postgres:postgres@localhost:5432/example?sslmode=disable
+```
+`sslmode=disable` means that the connection with out database will not be encrypted. Enabling it is left as an exercise.
+
+You can find further description of database URLs [here](README.md#database-urls).
+## Create migrations
+Let's create table called `users`:
+```
+migrate create -ext sql -dir db/migrations create_users_table
```

Comment:
I'd recommend using the `-seq` option

Reply:
I agree that it makes the example easier to understand but using `-seq` in a project used by many developers can go awry as they can push two migrations with the same sequential number. I feel like it's one of these "know what you're doing" options, while I wanted the tutorial to be accessible for someone who did not use database migrations before. What's your opinion here?

Reply:
Detecting conflicting migrations is an unresolved issue. Using sequentially numbered vs timestamps doesn't fix the issue. e.g. when using timestamps, 2 different developers could create conflicting migrations and the developer who created created the migration second gets their PR merged before the first one.

Could you also mention this caveat (that conflicting migrations aren't detected) in our docs.

For more info, see: https://github.com/golang-migrate/migrate/issues/179#issuecomment-475821264

Reply:
I have not thought about that. I will mention it in getting started docs.